### PR TITLE
fix: Avoids dumping zero values for Cluster electable specs in json that leads to API validation errors

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Proposed changes

Avoids dumping zero values in json that leads to API validation errors

Link to any related issue(s): CLOUDP-265793

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/cluster/cmd/resource/mappings_test.go [hunk 1]
```diff
@@ -79,3 +80,39 @@ func regionConfig(provider, region string) *[]admin.CloudRegionConfig {
 		ProviderName: &provider,
 	}}
 }
+
+func TestNewHardwareSpec(t *testing.T) {
+	testCases := map[string]struct {
+		expected string
```
Comment (AgustinBettati) [inline cfn-resources/cluster/cmd/resource/mappings_test.go:86]: q: is there a reason to not use admin.HardwareSpec for expected value? Just thinking adding the marshaling aspect makes it harder to maintain and not seeing a clear benefit.

File: cfn-resources/cluster/mongodb-atlas-cluster.json [hunk 2]
```diff
@@ -82,7 +82,7 @@
       "type": "object",
       "properties": {
         "DiskIOPS": {
-          "type": "string",
+          "type": "integer",
```
Comment (AgustinBettati) [inline cfn-resources/cluster/mongodb-atlas-cluster.json:85]: Is this change related or needed? While the change appears correct, this is a breaking change and will require updating CDK as well.

**Review by AgustinBettati — COMMENTED** [review]
Changes related to bug fix look good, have some doubt on the disk iops change

File: cfn-resources/cluster/cmd/resource/mappings_test.go (hunk 1 above)
Comment (EspenAlbert) [reply cfn-resources/cluster/cmd/resource/mappings_test.go:86]: More compact and tests the behavior of marshaling that the API call will use. But I'm open to changing it

File: cfn-resources/cluster/mongodb-atlas-cluster.json (hunk 2 above)
Comment (EspenAlbert) [reply cfn-resources/cluster/mongodb-atlas-cluster.json:85]: This is true. Thank you for calling it out.
I don't see a reason for why it is a string. Both resources in the TF provider use Integer.
Although it is a breaking change, it will only impact users upgrading the CDK/CFN resource, and the fix should be simple.

File: cfn-resources/cluster/cmd/resource/mappings_test.go (hunk 1 above)
Comment (AgustinBettati) [reply cfn-resources/cluster/cmd/resource/mappings_test.go:86]: > tests the behavior of marshaling that the API call will use

I believe this falls more in the scope of the SDK implementation, we have control over how we populate the SDK model and have understanding that if a nil value is provided for a pointer it is not sent in the request. Don't see this too relevant so adjust as you see fit.

File: cfn-resources/cluster/mongodb-atlas-cluster.json (hunk 2 above)
Comment (Zuhairahmed) [reply cfn-resources/cluster/mongodb-atlas-cluster.json:85]: suggest leave as is for now, if customer ever asks about it we can revisit. rather then forcing parity between the tools, once something is out customers in that community may already be used to it working this way so would lean on developer/customer empathy here in avoiding breaking changes when possible.

File: cfn-resources/cluster/cmd/resource/mappings_test.go (hunk 1 above)
Comment (EspenAlbert) [reply cfn-resources/cluster/cmd/resource/mappings_test.go:86]: choosing to keep the code as is. In this case we would be notified if the SDK would change unexpectedly.

File: cfn-resources/cluster/mongodb-atlas-cluster.json (hunk 2 above)
Comment (EspenAlbert) [reply cfn-resources/cluster/mongodb-atlas-cluster.json]: Reverted in be5fdfe361cfd5df527ca9418c4ccb3594c320c3 and re-implemented in conversion in d1c504df5ec4e0ca8dbaf3a4ac01014383114dab

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for followup changes and good conversations
